packages/flutter/lib/src/painting/image_resolution.dart

```diff
@@ -265,10 +265,13 @@ class AssetImage extends AssetBundleImageProvider {
       return _naturalResolution;
     }
 
-    final File assetPath = File(key);
-    final Directory assetDir = assetPath.parent;
+    final List<String> assetPathSegments = key.split(_kSeparator);
```

Comment:
Doing our own half-backed path parsing here is scary. If you are certain that we are dealing with an Uri, could you use Uri.parse and then extract what you want from that object?

Reply:
Done